scripts/lake-build-wrapper.py

```diff
@@ -0,0 +1,352 @@
+#!/usr/bin/env python3
+"""
+Usage: lake-build-wrapper.py <output_file.json> <command> [args...]
+
+Wrapper script for `lake build` to be used in GitHub actions that:
+1. Groups only normal build output (✔ lines) into collapsible log groups
+2. Emits all non-✔ lines (including ℹ/⚠/✖ and summaries) outside groups
+3. Extracts infos/warnings/errors to JSON output file
+
+Summary JSON format:
+- Top-level object with keys:
+  - warnings: list of issue objects (warning blocks)
+  - errors: list of issue objects (error blocks)
+  - infos: list of issue objects (info blocks)
+  - warning_count: integer
+  - error_count: integer
+  - info_count: integer
+
+- Each issue object contains:
+  - file_info: object with parsed progress/target info, if available
+    - current: integer (N in "[N/M]")
+    - total: integer (M in "[N/M]")
+    - target: string (module/target token from the log)
+    - file: string or null (converted Lean filename; null for non-file targets containing ':')
+  - messages: list of strings (subset of lines in the block that include typical markers like "warning:", "error:", or "info:")
+  - full_output: string (the entire contiguous block captured for the issue)
+
+State machine overview:
```

Comment:
This is a bit dry: can you elaborate a bit more what this state machine does/is used for?